translator/config/mode.go

```diff
@@ -8,10 +8,16 @@ const (
 	ModeOnPrem    = "onPrem"
 	ModeOnPremise = "onPremise"
 	ModeWithIRSA  = "withIRSA"
+	ModeEKS       = "EKS"
+	ModeK8sEC2    = "K8sEC2"
+	ModeK8sOnPrem = "K8sOnPrem"
```

Comment:
We have a few places in the code where we depend entirely on the mode being `onPrem`/`onPremise` and I'm concerned that this will cause a regression.

One option is to have a function that checks if the mode is an `onPrem` mode and include the `K8sOnPrem` as part of that and replace the existing checks.